projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -4884,6 +4905,136 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
+        if (rowIndex < 0 || rowIndex > this.verticalScrollContainer.igxForOf.length - 1
+                ||  (visibleColIndex !== -1 && this.columnList.map(col => col.visibleIndex).indexOf(visibleColIndex) === -1)) {
+            return;
+        }
+        this.wheelHandler();
+        if (visibleColIndex === -1 || (this.navigation.isColumnFullyVisible(visibleColIndex)
+                && this.navigation.isColumnLeftFullyVisible(visibleColIndex))) {
+            if (this.navigation.shouldPerformVerticalScroll(rowIndex)) {
+                this.verticalScrollContainer.scrollTo(rowIndex);
+                this.verticalScrollContainer.onChunkLoad
+                .pipe(first()).subscribe(() => {
+                    this.executeCallback(rowIndex, visibleColIndex, cb);
+                });
+            } else {
+                this.executeCallback(rowIndex, visibleColIndex, cb);
+            }
+        } else {
+            const unpinnedIndex = this.navigation.getColumnUnpinnedIndex(visibleColIndex);
+            this.parentVirtDir.onChunkLoad
+                .pipe(first())
+                .subscribe(() => {
+                    if (this.navigation.shouldPerformVerticalScroll(rowIndex)) {
+                        this.verticalScrollContainer.scrollTo(rowIndex);
+                        this.verticalScrollContainer.onChunkLoad
+                        .pipe(first()).subscribe(() => {
+                            this.executeCallback(rowIndex, visibleColIndex, cb);
+                        });
+                    } else {
+                        this.executeCallback(rowIndex, visibleColIndex, cb);
+                    }
+
+                });
+            this.navigation.horizontalScroll(rowIndex).scrollTo(unpinnedIndex);
+        }
+    }
+
+    public getNextCell(currRowIndex: number, curVisibleColIndex: number, callback: (IgxColumnComponent) => boolean = null) {
+        const columns = this.columnList.filter(col => !col.columnGroup && col.visibleIndex >= 0);
+
+        if (!this.isValidPosition(currRowIndex, curVisibleColIndex)) {
+            return {rowIndex: currRowIndex, visibleColumnIndex: curVisibleColIndex};
+        }
+        const colIndexes = callback ? columns.filter((col) => callback(col)).map(editCol => editCol.visibleIndex).sort((a, b) => a - b) :
+                                columns.map(editCol => editCol.visibleIndex).sort((a, b) => a - b);
+        const nextCellIndex = colIndexes.find(index => index > curVisibleColIndex);
+        if (this.verticalScrollContainer.igxForOf.slice(currRowIndex, currRowIndex + 1)
+                .find(rec => !rec.expression && !rec.summaries) && nextCellIndex !== undefined) {
+            return {rowIndex: currRowIndex, visibleColumnIndex: nextCellIndex};
+        } else {
+            if (colIndexes.length === 0 || this.getNextDataRowIndex(currRowIndex) === currRowIndex) {
+                return {rowIndex: this.getNextDataRowIndex(currRowIndex), visibleColumnIndex: curVisibleColIndex};
```

Comment:
Please introduce a new type/interface. We don't want anonymous objects. See ["Anonymous objects is a no-no."](https://github.com/IgniteUI/igniteui-angular/pull/4706#discussion_r282767504)